### PR TITLE
Implementation of ConvMixer

### PR DESCRIPTION
**Pull request by theabhirath**
This is an implementation of [ConvMixer](https://arxiv.org/abs/2201.09792). This is a _pretty_ small model (to code, params weigh in at a solid 52M), but has some interesting experiments done around it. Something I cooked up in between trying to make some larger models work out...

### CODE REVIEW

**Comment by theabhirath** [comment]
Bump?

File: src/convnets/convmixer.jl [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+([reference](https://arxiv.org/abs/2201.09792))
+
+# Arguments
+- `planes`: number of planes in the output of each block
+- `depth`: number of layers
+- `inchannels`: number of channels in the input
+- `kernel_size`: kernel size of the convolutional layers
+- `patch_size`: size of the patches
+- `activation`: activation function used after the convolutional layers
+- `nclasses`: number of classes in the output
```
Comment (CarloLucibello) [inline src/convnets/convmixer.jl:48]: these do not correspond to the signature

File: src/layers/conv.jl [hunk 2]
```diff
@@ -34,6 +35,10 @@ function conv_bn(kernelsize, inplanes, outplanes, activation = relu;
     bnplanes = outplanes
   end
 
+  if preact
+    activations = (conv = activation, bn = identity)
+  end
```
Comment (CarloLucibello) [inline src/layers/conv.jl:40]: are all 4 combinations of the `rev` and `preact` working as they are supposed to? just checking

**Comment by CarloLucibello** [comment]
needs a mention of ConvMixer in the docs (which is just the README?)

File: src/convnets/convmixer.jl (hunk 1 above)
Comment (theabhirath) [reply src/convnets/convmixer.jl]: Fixed now!

File: src/layers/conv.jl (hunk 2 above)
Comment (theabhirath) [reply src/layers/conv.jl:41]: There was one edge case not being handled, I think - should be working now

File: src/layers/conv.jl [hunk 3]
```diff
@@ -12,6 +12,7 @@ Create a convolution + batch normalization pair with ReLU activation.
 - `outplanes`: number of output feature maps
 - `activation`: the activation function for the final layer
 - `rev`: set to `true` to place the batch norm before the convolution
+- `preact`: set to `true` to place the activation function before the batch norm
```
Comment (darsnack) [inline src/layers/conv.jl:15]:
```suggestion
- `preact`: set to `true` to place the activation function before the batch norm
            (only compatible with `rev = false`)
```

**Review by darsnack — CHANGES_REQUESTED** [review]
One small suggestion. Looks good otherwise!

**Review by darsnack — APPROVED** [review]

**Pull request merged**